app/metadata.js

```diff
@@ -1,79 +1,109 @@
-const statsdClient = require('./statsd');
 const domino = require('domino');
+const fastimage = require('fastimage');
+const statsdClient = require('./statsd');
 const {getMetadata, makeUrlAbsolute} = require('page-metadata-parser');
 require('isomorphic-fetch');
 
-function getDocumentMetadata(url, window) {
-  const doc = window.document;
-  const metadata = getMetadata(doc, url);
-
-  const responseData = {
-    url: metadata.url,
-    original_url: url,
-    title: metadata.title,
-    description: metadata.description,
-    favicon_url: metadata.icon_url || makeUrlAbsolute(url, '/favicon.ico'),
-    images: []
-  };
+function fetchUrlContent(url) {
+  const startFetch = statsdClient.getTimestamp();
 
-  if (metadata.image_url) {
-    responseData.images = [{
-      url: metadata.image_url,
-      width: 500,
-      height: 500,
-      entropy: 1.0,
-    }];
-  }
+  return fetch(url).then(res => {
+    const endFetch = statsdClient.getTimestamp();
+    statsdClient.timing('fetch_time', (endFetch - startFetch));
 
-  return responseData;
+    if (res.status >= 200 && res.status < 300) {
+      statsdClient.increment('fetch_success');
+      return res;
+    } else {
+      throw new Error(`Request Failure: ${res.status} ${res.statusText}`);
+    }
+  }).then(res => res.text())
+  .catch(e => {
+    statsdClient.increment('fetch_fail');
+
+    const endFetch = statsdClient.getTimestamp();
+    statsdClient.timing('fetch_time', (endFetch - startFetch));
+
+    throw e;
+  });
 }
 
+function getHtmlDocument(html) {
+  return domino.createWindow(html).document;
+}
+
+function getImageInfo(imageUrl) {
+  const startImage = statsdClient.getTimestamp();
+
+  return fastimage.info(imageUrl).then(info => {
+    const endImage = statsdClient.getTimestamp();
+    statsdClient.timing('image_time', (endImage - startImage));
+    statsdClient.increment('image_success');
+
+    return info;
+  }).catch(e => {
+    const endImage = statsdClient.getTimestamp();
+    statsdClient.timing('image_time', (endImage - startImage));
+    statsdClient.increment('image_fail');
+
+    throw e;
+  });
+}
+
+function getDocumentMetadata(url, doc) {
+  return new Promise(resolve => {
+    const parsedMetadata = getMetadata(doc, url);
+
+    const urlMetadata = {
+      url: parsedMetadata.url,
+      original_url: url,
+      title: parsedMetadata.title,
+      description: parsedMetadata.description,
+      favicon_url: parsedMetadata.icon_url || makeUrlAbsolute(url, '/favicon.ico'),
```

Comment:
@pdehaan An update to the latest metadata parser will fix this in a separate PR.